ui/src/main/java/de/ipb_halle/lbac/admission/PluginSettingsDialogControlerBean.java

```diff
@@ -0,0 +1,372 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package de.ipb_halle.lbac.admission;
+
+import java.io.Serializable;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.faces.view.ViewScoped;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import de.ipb_halle.lbac.i18n.UIMessage;
+
+/**
+ * This JSF backing bean controls the form of the plugin user settings.
+ * <p>
+ * View file: WEB-INF/templates/accountSettings/pluginsTab.xhtml
+ *
+ * @author flange
+ */
+@ViewScoped
+@Named
+public class PluginSettingsDialogControlerBean implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final static String MESSAGE_BUNDLE = "de.ipb_halle.lbac.i18n.messages";
+
+    @Inject
+    private UserBean userBean;
+
+    private Logger logger;
+
+    private String molPluginType;
+
+    private String previewStructure;
+
+    /**
+     * default constructor
+     */
+    public PluginSettingsDialogControlerBean() {
+        this.logger = LogManager.getLogger(this.getClass().getName());
+    }
+
+    /**
+     * Initializes the bean state by settings the plugin types the their
+     * preferred value and the preview chemical structure to benzene.
+     */
+    @PostConstruct
+    public void init() {
+        molPluginType = userBean.getPluginSettings()
+                .getPreferredMolPluginType();
+        previewStructure = benzene;
+    }
+
+    public String getMolPluginType() {
+        return molPluginType;
+    }
+
+    public void setMolPluginType(String molPluginType) {
+        this.molPluginType = molPluginType;
+    }
+
+    public List<String> getAvailableMolPluginTypes() {
+        return userBean.getPluginSettings().getAllMolPluginTypes();
+    }
+
+    public String getPreviewStructure() {
+        return previewStructure;
+    }
+
+    public void setPreviewStructure(String previewStructure) {
+        // do nothing
+    }
+
+    /**
+     * Saves the selected plugin types as preferred plugin types and notifies to
+     * user upon success.
+     */
+    public void actionSave() {
+        User currentUser = userBean.getCurrentAccount();
+
+        if (!currentUser.isPublicAccount()) {
+            if (currentUser
+                    .getSubSystemType() == AdmissionSubSystemType.LOCAL) {
+                userBean.getPluginSettings()
+                        .setPreferredMolPluginType(molPluginType);
+
+                UIMessage.info(MESSAGE_BUNDLE, "admission_plugins_updated");
```

Comment:
Please extract the MessagePresenter.
Example in:
  ExperimentBean (messagePresenter)